scripts/view-item-info.lua

```diff
@@ -0,0 +1,380 @@
+        append(list)
+        return list
+    end
+    add_lines_to_list(list, get_plant_reaction_products(mat))
+    if item._type == df.item_plantst and GetMatPlant (item) then
+        local plant = GetMatPlant (item)
+        for k,v in pairs (plant.material_defs) do
+            if v ~= -1 and string.find (k,"type_") and not string.find (k,"type_basic")
+                    or string.find (k,"type_seed") or string.find (k,"type_tree") then
+                local targetmat = dfhack.matinfo.decode (v,
+                    plant.material_defs["idx_"..string.match (k,"type_(.+)")])
+                local state = "Liquid"
+                if string.find (k,"type_mill") then state = "Powder"
+                elseif string.find (k,"type_thread") then state = "Solid" end
+                local st_name = targetmat.material.state_name[state]
+                append(list,"Used to make "..targetmat.material.prefix..''..st_name)
+            end
+        end
+    end
+    append(list)
+    return list
+end
+
+function get_all_uses_strings (item)
+    local all_lines = {}
+    local FoodsAndPlants = {df.item_meatst, df.item_globst,
+        df.item_plantst, df.item_plant_growthst, df.item_liquid_miscst,
+         df.item_powder_miscst, df.item_cheesest, df.item_foodst}
+    local ArmourTypes = {df.item_armorst, df.item_pantsst,
+        df.item_helmst, df.item_glovesst, df.item_shoesst}
+    if df.global.gamemode == df.game_mode.ADVENTURE and item ~= "COIN" then
+        add_lines_to_list(all_lines, {{"Value: "..dfhack.items.getValue(item),0}})
+    elseif isInList(ArmourTypes, item._type) then
+        add_lines_to_list(all_lines, GetArmorPropertiesStringList(item))
+    elseif item._type == df.item_weaponst or item._type == df.item_toolst then
+        add_lines_to_list(all_lines, GetWeaponPropertiesStringList(item))
+    elseif item._type == df.item_ammost then
+        add_lines_to_list(all_lines, GetAmmoPropertiesStringList(item))
+    elseif item._type == df.item_shieldst then
+        add_lines_to_list(all_lines, GetShieldPropertiesStringList(item))
+    elseif item._type == df.item_seedsst then
+        local str = math.floor(GetMatPlant(item).growdur/12).." days"
+        add_lines_to_list(all_lines, {{"Growth time: "..str, 0}})
+    elseif isInList(FoodsAndPlants, item._type) then
+        add_lines_to_list(all_lines, GetFoodPropertiesStringList(item))
+    end
+    if not dfhack.items.isCasteMaterial(df.item_type[get_textid(item)]) then
+        add_lines_to_list(all_lines, GetMatPropertiesStringList(item))
+    end
+    return all_lines
+end
+
+function get_custom_item_desc (item)
+    local desc
+    local ID = get_textid (item)
+    if ID and dfhack.items.getSubtypeCount(df.item_type[ID]) ~= -1 then
+        ID = item.subtype.id end
+    if not ID then return nil end
+    if dfhack.findScript("item-descriptions") then
+        local desc = dfhack.script_environment("item-descriptions").descriptions.ID
```

Comment:
Out of curiosity, did you actually test this? I had to fix a couple of issues with this section for it to work, namely:
* This should be using `descriptions[ID]` instead. `descriptions.ID` is equivalent to `descriptions["ID"]`.
* Declaring `desc` as a local variable here scopes it to the `if` block only - `desc` will be `nil` anywhere besides this line.